pkg/cmd/server/server.go

```diff
@@ -59,9 +58,12 @@ var ServerCmd = &cobra.Command{
 		select {
 		case err := <-errCh:
 			log.Fatal(err)
-		// TODO: This is an optimistic check. We should check if the server is actually running
-		case <-time.After(5 * time.Second):
-			printServerStartedMessage(c)
+		default:
```

Comment:
@tarunrajput thanks for the PR!

With this `default` case, you ignored the `errCh` above which will pass errors from the `Start` function. This can lead to situations where the server start errored but you did not catch it.